### DISCUSSION TITLE
background-hooks script is sending two messages to content-hooks when it is expected to send one

### DISCUSSION THREAD

**Discussion by daveavi** [quasarframework/quasar#8843] [CLI - BEX mode]
Hello! 

Currently, I am writing a basic Quasar BEX application to get myself more familiar with the framework.  In my background-hooks script, I have an on-click listener that activates/deactivates my Quasar app accordingly. I am running into an issue where when I click on my extension icon once to remove my Qusar app off the web page, it sends two messages, where it removes the Quasar app but it brings it back up again. It looks like my background- hooks script is sending the message twice off one click and I am not sure why. Here is the code:

![image](https://user-images.githubusercontent.com/20130700/114275771-c834be80-99f1-11eb-8549-05f78f4154ee.png)

I would love to know why this behavior is occurring so I can have my Quasar App open/close normally.

Thanks,

Avi Dave

**Comment by daveavi** [comment]
Sometimes this is also sending multiple messages. I am running into similar issues while running https://github.com/smolinari/quasar-bex-todo-app.

**Reply by webnoob** [reply]
Can you show all your code please? It looks as though the event listener is being added multiple times.

**Reply by daveavi** [reply]
Here is my background-hooks.js : 

![image](https://user-images.githubusercontent.com/20130700/114739914-5624e800-9d17-11eb-962b-6f8bcf52e7ee.png)

![image](https://user-images.githubusercontent.com/20130700/114739940-5cb35f80-9d17-11eb-8059-77d88024c0da.png)

Here is my Vue Code: 
![image](https://user-images.githubusercontent.com/20130700/114739983-69d04e80-9d17-11eb-8206-2d2464340ce0.png)

**Reply by chenlevy** [reply]
Same issue here, did you find a solution?
Sometimes it calls the background listener function multiple times - sometimes as many times as the number of open tabs running the extension.

Running quasar 1.15.2

**Reply by daveavi** [reply]
Sorry for the late reply, but I had to move away from quasar for this reason. I couldn't seem to understand what was going wrong, so I decided to just go with the familiar way to build chrome extensions. I am still using Vue, but instead of using a framework, I used laravel mix for bundling my vue and sass code

**Comment by ArsalaBangash** [comment]
@daveavi I would appreciate it if you could link your repository, and write out the steps to reproduce this issue. 

Furthermore, I saw that you used quasar V1, so I created [a merge request on GitLab](https://gitlab.com/grey-software/focused-browsing/-/merge_requests/1) where I added a prototype so you can try and see if this issue still exists on V2. 

Finally, I went ahead and recorded [a video of how the extension behaves on my personal computer](https://www.loom.com/share/75ddbf2101e5416fbe0b6c30d1417ea3) from the latest master commit on
 [your repo](https://github.com/daveavi/quasar-bex-tutorial)

**Reply by ArsalaBangash** [reply]
https://www.loom.com/share/75ddbf2101e5416fbe0b6c30d1417ea3

**Reply by daveavi** [reply]
Here is my repo: https://github.com/daveavi/quasar-bex-tutorial 

To reproduce the error, here are the command line steps:
1. cd my-bex-project
2. Make sure you do npm install or yarn to download any dependencies
3. quasar dev -m bex, to enter bex mode of the app and building our developer version of our application

Once you have done these commands, go to chrome and on the toolbar, you want to click on the extension's icon, and then manage extensions:
![image](https://user-images.githubusercontent.com/20130700/114882396-e9baef00-9dd1-11eb-98e2-bf73f1d02c33.png)

You then want to enter developer mode, and click on load unpacked , to load your extension:
<img width="216" alt="Screen Shot 2021-04-15 at 10 04 16 AM" src="https://user-images.githubusercontent.com/20130700/114882534-0c4d0800-9dd2-11eb-82cc-0678c2f509b3.png">
<img width="138" alt="Screen Shot 2021-04-15 at 10 04 22 AM" src="https://user-images.githubusercontent.com/20130700/114882546-0e16cb80-9dd2-11eb-9562-df7e064e60a3.png">

Here you want to choose your src-bex folder. Once the extension is loaded, make sure to pin the extension, open a new tab, can be any website. Click on the extension icon, initially it should load the quasar app, but try to click it again, and it will still stay on the page. 

It is demonstrated in this video right here: https://www.loom.com/share/75ddbf2101e5416fbe0b6c30d1417ea3

**Reply by daveavi** [reply]
After running the extension on V2, there is still issues with the event listeners: 
https://www.loom.com/share/8022badfeec242ab9f9b0425b68551d8 


Here are some logs that represents the toggle functionality: 
![image](https://user-images.githubusercontent.com/20130700/114890459-06a6f080-9dd9-11eb-96a6-9e87f0abff50.png)

**Reply by daveavi** [reply]
@webnoob I hope you are doing well! I would love to know if you have any insight on this issue as this issue is blocking me from using Quasar to build a browser extension.

**Comment by swtanggara** [comment]
Hi @daveavi I had same issue in my first try using BEX Mode in Quasar. But then I've managed to trick it using this code:

```js
function defaultThrottle(throttled) {
  return throttle(throttled, 100, { trailing: false });
}

this.$q.bex.on('bex.toggle.toolbar', defaultThrottle(this.toggleToolbar));
```

**Reply by volksec** [reply]
up
